File: src/main/java/com/partymakers/shareparty/domain/party/usecase/impl/GetPartyFriendsImpl.java
package com.partymakers.shareparty.domain.party.usecase.impl;

import com.partymakers.shareparty.domain.friends.entity.Friend;
import com.partymakers.shareparty.domain.party.port.PartyRoomRepository;
import com.partymakers.shareparty.domain.party.usecase.GetPartyFriends;
import com.partymakers.shareparty.domain.party.usecase.exception.NotFoundException;

import java.util.List;
import java.util.Set;

import lombok.RequiredArgsConstructor;

@RequiredArgsConstructor
public class GetPartyFriendsImpl implements GetPartyFriends {

    private final PartyRoomRepository repository;

    @Override
    public Set<Friend> getPartyFriends(long partyId) {
        return repository.findById(partyId).map(partyRoom -> partyRoom.getFriends())
            .orElseThrow(NotFoundException::new);
    }
}

File: src/main/kotlin/com/partymakers/shareparty/domain/party/usecase/impl/GetPartyFriendsImpl.kt
package com.partymakers.shareparty.domain.party.usecase.impl

import com.partymakers.shareparty.domain.friends.entity.Friend
import com.partymakers.shareparty.domain.party.port.PartyRoomRepository
import com.partymakers.shareparty.domain.party.usecase.GetPartyFriends
import com.partymakers.shareparty.domain.party.usecase.exception.NotFoundException
import org.springframework.stereotype.Service

@Service
class GetPartyFriendsImpl(
    private val repository: PartyRoomRepository
) : GetPartyFriends {

    override fun getPartyFriends(partyId: Long): Set<Friend> =
        repository.findById(partyId)
            .map { it.friends }
            .orElseThrow { NotFoundException("Party room with id: $partyId not found") }
} 